### PR TITLE
DOCSP-41121 Connect to Mongo Client 

### PR DESCRIPTION
**Pull request by lindseymoore**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-kotlin-sync/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-41121>
Staging - <https://preview-mongodblindseymoore.gatsbyjs.io/kotlin-sync/DOCSP-41121-create-mongoclient/connect/mongoclient/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/connect/mongoclient.txt [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+and verify that the connection is successful:
+
+.. literalinclude:: /includes/connect/mongoclient.kt
+   :start-after: start-connect-to-atlas
+   :end-before: end-connect-to-atlas
+   :language: kotlin
+   :copyable:
+   :dedent:
+
+Connect to a MongoDB Server or Replica Set
```
Comment (mcmorisi) [inline source/connect/mongoclient.txt:116]: I think everything in this section can be removed – it overlaps with the [Choose a Connection Target](https://github.com/mongodb/docs-kotlin-sync/pull/33) page currently in PR. I think this page should strictly cover creating a client, while the Choose a Connection Target page contains details about the different types of MongoDB instances you can connect to.

File: source/connect/mongoclient.txt [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+------------------------
+
+To connect to a MongoDB deployment on Atlas, create a client. You can create a 
+client that uses your connection string and other client options by passing a 
+``MongoClientSettings`` object to the ``MongoClient.create()`` method.
+
+To instantiate a ``MongoClientSettings`` object, use the builder method to 
+specify your connection string and any other client options, and then call 
+the ``build()`` method. Chain the ``applyConnectionString()``` method to the 
+builder to specify your connection URI.
```
Comment (mcmorisi) [inline source/connect/mongoclient.txt:99]:
```suggestion
To instantiate a ``MongoClientSettings`` object, use the builder method to 
specify your connection string and any other client options by chaining the appropriate methods.
Once you have set your desired configurations, call the ``build()`` method.
```

File: source/connect/mongoclient.txt [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+       this component, the client authenticates the user against the ``admin`` database.
+
+   * - ``?<options>``
+
+     - Optional. A query string that specifies connection-specific
+       options as ``<name>=<value>`` pairs. See
+       :ref:`kotlin-sync-connection-options` for a full description of
+       these options.
+
+.. Is there relevant connection string content for kotlin sync to ref to? Only see for java sync.
```
Comment (mcmorisi) [inline source/connect/mongoclient.txt:83]: I would confirm with the Java DBX team but it's usually the case that Java and Kotlin content overlaps heavily.

When in doubt, we can always link readers to the [Server manual](https://www.mongodb.com/docs/manual/reference/connection-string/) for more general-purpose reading on a subject.

```suggestion
```

**Review by mcmorisi — CHANGES_REQUESTED** [review]
A few things!

File: source/connect/mongoclient.txt [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+For more information about creating a connection string, see
+:manual:`Connection Strings </reference/connection-string?tck=docs_driver_kotlin>` in the
+MongoDB Server documentation.
+
+Atlas Connection Example
+------------------------
+
+To connect to a MongoDB deployment on Atlas, create a client. You can create a 
+client that uses your connection string and other client options by passing a 
+``MongoClientSettings`` object to the ``MongoClient.create()`` method.
```
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/connect/mongoclient.txt:94]: Good revision, thanks!

File: source/connect/mongoclient.txt (hunk 2 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt:99]: Revised!

**Review by mcmorisi — APPROVED** [review]
LGTM!

File: source/connect/mongoclient.txt (hunk 3 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt]: Hi @jyemin, do you know if the connection string information for java sync is applicable for kotlin sync? I would like to link to connection string information on this page. This is the connection string page for java sync: https://www.mongodb.com/docs/manual/reference/connection-string/?tck=docs_driver_python

Comment (jyemin) [reply source/connect/mongoclient.txt]: Yes, connection string info is exactly the same.

File: source/connect/mongoclient.txt [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+to connect to MongoDB.
+
+.. _kotlin-sync-connection-uri:
+
+Connection URI
+--------------
+
+A standard connection string includes the following components:
+
+.. TODO, add this as last sentence for ``username:password`` description: For more information about the ``authSource``
```
Comment (jyemin) [inline source/connect/mongoclient.txt:43]: Does this TODO need to be done?

File: source/connect/mongoclient.txt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+Connection URI
+--------------
+
+A standard connection string includes the following components:
+
+.. TODO, add this as last sentence for ``username:password`` description: For more information about the ``authSource`` 
+.. connection option, see :ref:`kotlin-sync-auth`. 
+
+.. list-table::
```
Comment (jyemin) [inline source/connect/mongoclient.txt:46]: Are these all a direct copy from Java driver?  Do we want to duplicate it like this?  Up to you but seems hard to maintain.

File: source/connect/mongoclient.txt [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+     - Optional. The authentication database to use if the
+       connection string includes ``username:password@``
+       authentication credentials but not the ``authSource`` option. If you don't include 
+       this component, the client authenticates the user against the ``admin`` database.
+
+   * - ``?<options>``
+
+     - Optional. A query string that specifies connection-specific
+       options as ``<name>=<value>`` pairs. See
+       :ref:`kotlin-sync-connection-options` for a full description of
```
Comment (jyemin) [inline source/connect/mongoclient.txt:79]: Is this page already live?  If not, when will it be added?

File: source/connect/mongoclient.txt [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+   :class: singlecol
+
+Overview
+--------
+
+To connect to a MongoDB deployment, you need two things:
+
+- A **connection URI**, also known as a *connection string*, which tells the {+driver-short+}
+  which MongoDB deployment to connect to.
+- A **MongoClient** object, which creates the connection to and perform
```
Comment (jyemin) [inline source/connect/mongoclient.txt:27]: perform -> performs

File: source/connect/mongoclient.txt [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+--------
+
+To connect to a MongoDB deployment, you need two things:
+
+- A **connection URI**, also known as a *connection string*, which tells the {+driver-short+}
+  which MongoDB deployment to connect to.
+- A **MongoClient** object, which creates the connection to and perform 
+  operations on the MongoDB deployment.
+
+You can also use either of these components to customize the way the {+driver-short+} behaves
```
Comment (jyemin) [inline source/connect/mongoclient.txt:30]: You can't really use MongoClient to customize.  You can use MongoClientSettings though.

File: source/connect/mongoclient.txt [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+MongoDB Server documentation.
+
+Atlas Connection Example
+------------------------
+
+To connect to a MongoDB deployment on Atlas, you must first create a client. 
+You can create a client with your desired configurations by passing a 
+``MongoClientSettings`` object to the ``MongoClient.create()`` method.
+
+To instantiate a ``MongoClientSettings`` object, use the builder method to
```
Comment (jyemin) [inline source/connect/mongoclient.txt:93]: Should we also say that you can create the MongoClient directly from a connection string?

**Review by jyemin — CHANGES_REQUESTED** [review]

File: source/connect/mongoclient.txt (hunk 5 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt:43]: A reminder for when the kotlin authentication page is created. Made comment clearer, thanks!

File: source/connect/mongoclient.txt (hunk 6 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt:46]: Separate sites are maintained for each driver as of right now. For the Kotlin Sync work, we are using a combination of the [PyMongo](https://www.mongodb.com/docs/languages/python/pymongo-driver/current/connect/mongoclient/) and [Kotlin Coroutine](https://www.mongodb.com/docs/drivers/kotlin/coroutine/current/fundamentals/connection/connect/) docs as a model.

File: source/connect/mongoclient.txt (hunk 7 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt:79]: As of right now, this links to a placeholder page for connection options. The page should be done in another few weeks.

File: source/connect/mongoclient.txt (hunk 10 above)
Comment (lindseymoore) [reply source/connect/mongoclient.txt:104]: Added!

File: source/connect/mongoclient.txt [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+
+   * - ``?<options>``
+
+     - Optional. A query string that specifies connection-specific
+       options as ``<name>=<value>`` pairs. See
+       :ref:`kotlin-sync-connection-options` for a full description of
+       these options.
+
+For more information about creating a connection string, see
+:manual:`Connection Strings </reference/connection-string?tck=docs_driver_java_sync>` in the
```
Comment (lindseymoore) [inline source/connect/mongoclient.txt:83]:
```suggestion
:manual:`Connection Strings </reference/connection-string?tck=docs_driver_java_sync>` in the
```
Is the ``tck`` correct for the java sync language in the selector? @mcmorisi

**Review by jyemin — APPROVED** [review]
LGTM!

File: source/connect/mongoclient.txt (hunk 11 above)
Comment (mcmorisi) [reply source/connect/mongoclient.txt:83]: Java Sync is probably the closest match but it's still technically a different language (despite sharing a lot of syntax with Kotlin). I see that there's no option in the dropdown for kotlin, so I think we can remove the `?tck=docs_driver_java_sync`

**Review by mcmorisi — APPROVED** [review]
LGTM!

**Pull request merged**